op-e2e/actions/action.go

```diff
@@ -0,0 +1,62 @@
+package actions
+
+import (
+	"context"
+
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils"
+)
+
+// Testing is an interface to Go-like testing,
+// extended with a context getter for the test runner to shut down individual actions without interrupting the test,
+// and a signaling function for when an invalid action is hit.
+// This helps custom test runners navigate slow or invalid actions, e.g. during fuzzing.
+type Testing interface {
+	e2eutils.TestingBase
+	// Ctx shares a context to execute an action with, the test runner may interrupt the action without stopping the test.
+	Ctx() context.Context
+	// InvalidAction indicates the failure is due to action incompatibility, does not stop the test.
+	InvalidAction(format string, args ...any)
+}
+
+// Action is a function that may change the state of one or more actors or check their state.
+// Action definitions are meant to be very small building blocks,
+// and then composed into larger patterns to write more elaborate tests.
+type Action func(t Testing)
+
+// ActionStatus defines the state of an action, to make a basic distinction between InvalidAction() and other calls.
+type ActionStatus uint
```

Comment:
Will there be more `ActionStatus` types than just `Ok` or `Invalid`

Reply:
I've a bunch of ideas; rewinding an action, skipping, or doing other special things for fuzzing. For now this is all I need though.